### PR TITLE
[NO_REVIEW] JAMES-2375 DSNBounce mailet do not provide a subject

### PR DESCRIPTION
**Pull request by chibenwa**
Subject empty fixed, Sender empty when you do not provide sender
property fixed too

See original PR: https://github.com/apache/james-project/pull/162

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java [hunk 1]
```diff
@@ -220,8 +220,7 @@ public String getMessage(Mail originalMail) throws MessagingException {
 
     @Override
     public Optional<MailAddress> getSender() throws MessagingException {
-        return SpecialAddressesUtils.from(this)
-                .getFirstSpecialAddressIfMatchingOrGivenAddress(getInitParameters().getSender(), RedirectNotify.SENDER_ALLOWED_SPECIALS);
+        return SpecialAddressesUtils.from(this).getFirstSpecialAddressIfMatchingOrGivenAddress(Optional.of(getInitParameters().getSender().orElse("postmaster")), RedirectNotify.SENDER_ALLOWED_SPECIALS);
```
Comment (mbaechler) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java]: this line is too long

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/DSNBounceTest.java [hunk 2]
```diff
@@ -523,7 +526,121 @@ public void dsnBounceShouldAddPrefixToSubjectWhenPrefixIsConfigured() throws Exc
                 .build();
 
         dsnBounce.service(mail);
+        
 
-        assertThat(mail.getMessage().getSubject()).isEqualTo("pre My subject");
+        List<SentMail> sentMails = fakeMailContext.getSentMails();
+        assertThat(sentMails).hasSize(1);
+        SentMail sentMail = sentMails.get(0);
+        assertThat(sentMail.getSubject().orElse(null)).isEqualTo("pre My subject");
```
Comment (mbaechler) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/DSNBounceTest.java]: orElse in an assertion?

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java (hunk 1 above)
Comment (chibenwa) [reply server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java]: Please comment in the original pull request in the description. Not my work here.

**Comment by chibenwa** [comment]
Underlying PR was merged

**Pull request closed without merging**